src/js/api/posts/create.mjs

```diff
@@ -0,0 +1,30 @@
+// Functionality to create a post
+
+import { authFetch } from "../authFetch.mjs";
+import { apiSocialUrl } from "../constants.mjs";
+
+const action = "/posts";
+const method = "post";
+
+export async function createPost(postData) {
+    const createPostUrl = apiSocialUrl + action;
+
+    try {
+        const response = await authFetch(createPostUrl, {
+            method,
+            body: JSON.stringify(postData)
+        });
+        if (response.ok) {
+            location.reload();
```

Comment:
Reloading the page stops JS from continuing, so your return never gets a chance to run. Consider redirecting in a UI function instead of an API function.